R/utilities-simulation.R

```diff
@@ -168,36 +168,25 @@ runSimulationsConcurrently <- function(simulations, simulationRunOptions = NULL,
   }
 
   simulations <- c(simulations)
-  # Create an Id <-> simulation map to get the correct simulation for the results.
-  simulationsIdMap <- list()
+  # Map of simulations ids to simulations objects
+  simulationIdSimulationMap <- vector("list", length(simulations))
+
+  # Add simulations
+  for (simulationIdx in seq_along(simulations)) {
+    simulation <- simulations[[simulationIdx]]
+    simulationIdSimulationMap[[simulationIdx]] <- simulation
+    names(simulationIdSimulationMap)[[simulationIdx]] <- simulation$id
 
-  # Create SimulationRunnerConcurrentOptions and add all simulations
-  for (simulation in simulations) {
-    simulationsIdMap[[simulation$id]] <- simulation
     rClr::clrCall(simulationRunner, "AddSimulation", simulation$ref)
   }
   # Run all simulations
   results <- rClr::clrCall(simulationRunner, "RunConcurrently")
-  # Pre-allocate lists for SimulationResult
-  simulationResults <- vector("list", length(simulations))
-  # Set the order of IDs so the results appear in the same order as simulations were provided
-  names(simulationResults) <- names(simulationsIdMap)
 
-  for (i in seq_along(results)) {
-    resultObject <- results[[i]]
-    id <- rClr::clrGet(resultObject, "Id")
-    succeeded <- rClr::clrGet(resultObject, "Succeeded")
-    if (succeeded) {
-      # Get the correct simulation and create a SimulationResults object
-      simulationResults[[id]] <- SimulationResults$new(ref = rClr::clrGet(resultObject, "Result"), simulation = simulationsIdMap[[id]])
-      next()
-    }
-    # If the simulation run failed, show a warning
-    if (!silentMode) {
-      errorMessage <- rClr::clrGet(resultObject, "ErrorMessage")
-      warning(errorMessage)
-    }
-  }
+  # Ids of the results are Ids of the simulations

```

Comment:
The logic for converting results of concurrent runner to `SimulationResults` is moved to a separate function as it is also used by `runSimulationBatches`.

Reply:
nice